src/main/java/org/springframework/retry/backoff/ExponentialBackOffPolicy.java

```diff
@@ -252,6 +269,18 @@ public void backOff(BackOffContext backOffContext) throws BackOffInterruptedExce
 		}
 	}
 
+	/**
+	 * Setter for {@link Log}. If not applied the following is used:
+	 * <p>
+	 * {@code LogFactory.getLog(getClass())}
+	 * </p>
+	 * @param logger the logger the exponential backoff policy uses for logging
+	 * @since 2.0.x
+	 */
+	public void setLogger(Log logger) {
```

Comment:
This is not correct. Such a functionality must not be exposed to end-user.
Please, consider to use a reflection in your tests instead.
See `DirectFieldAccessor`, for example.

Reply:
@artembilan Thanks for the insightful feedback about DirectFieldAccessor! I learned a lot about the importance of not polluting the public API with test-specific methods. Really appreciate the learning opportunity!